<?xml version="1.0"?>
<robot name="object"> <!-- simple puzzle robot-->
  <link name="base_link"> <!-- required name-->
    <inertial>
      <origin xyz="0 0 0" /> 
      <mass value="1.0" />
      <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="1.0"  iyz="0.0"  izz="100.0" />
    </inertial>
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.1 0.1 0.5" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.1 0.1 0.5" />
      </geometry>
    </collision>
  </link>
  
   
  <joint name="j_object_one" type="fixed">
    <parent link="base_link"/>
    <child link="l_object_one"/>
    <origin xyz="0 0 0.25"/>
  </joint>
        
  <link name="l_object_one"> <!-- upper part -->
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="1.0" />
      <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="100.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="0 0.3 0.05"/>
      <geometry>
        <box size="0.1 0.7 0.1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0.3 0.05"/>
      <geometry>
        <box size="0.1 0.7 0.1" />
      </geometry>
    </collision>
  </link>
  
  
  <joint name="j_object_two" type="fixed">
    <parent link="base_link"/>
    <child link="l_object_two"/>
    <origin xyz="0 0 -0.25"/>
  </joint>
        
  <link name="l_object_two"> <!-- lower part -->
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="1.0" />
      <inertia  ixx="100.0" ixy="0.0"  ixz="0.0"  iyy="1.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="-0.3 0 -0.05"/>
      <geometry>
        <box size="0.7 0.1 0.1"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="-0.3 0 -0.05"/>
      <geometry>
        <box size="0.7 0.1 0.1"/>
      </geometry>
    </collision>
  </link>
  
  <joint name="j_object_three" type="fixed">
    <parent link="base_link"/>
    <child link="l_object_three"/>
    <origin xyz="0 0 0.25"/>
  </joint>
        
  <link name="l_object_three"> <!-- upper second part / fourth branch -->
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="1.0" />
      <inertia  ixx="100.0" ixy="0.0"  ixz="0.0"  iyy="1.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="0.35 0.6 0.05"/>
      <geometry>
        <box size="0.6 0.1 0.1"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0.35 0.6 0.05"/>
      <geometry>
        <box size="0.6 0.1 0.1"/>
      </geometry>
    </collision>
  </link>
  
  <joint name="j_object_four" type="fixed">
    <parent link="base_link"/>
    <child link="l_object_four"/>
    <origin xyz="0 0 0.25"/>
  </joint>
        
  <link name="l_object_four"> <!-- upper second part / fiveth branch -->
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="1.0" />
      <inertia  ixx="100.0" ixy="0.0"  ixz="0.0"  iyy="1.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="0.6 0.35 0.05"/>
      <geometry>
        <box size="0.1 0.6 0.1"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0.6 0.35 0.05"/>
      <geometry>
        <box size="0.1 0.6 0.1"/>
      </geometry>
    </collision>
  </link>
</robot>
